2048.c

```diff
@@ -58,8 +58,7 @@ void drawBoard(uint8_t board[SIZE][SIZE]) {
 			if (board[x][y]!=0) {
 				char s[8];
 				snprintf(s,8,"%u",(uint32_t)1<<board[x][y]);
-				uint8_t t;
-				t=7-strlen(s);
+				uint t=7-strlen(s);
```

Comment:
why is this change needed?

Reply:
This seems to be a `-std=c90` error, not present while compiling with `-std=c99`

Reply:
I changed this line, because it was at her that gcc2 swore at Haiku. In any case, I have already changed it back, otherwise the assembly under the FreeBSD breaks down. I will try to compile under the haiku with the c99 parameter